src/Context/FeatureContext.php

```diff
@@ -868,6 +883,26 @@ public function create_db() {
 		self::run_sql( 'mysql --no-defaults', [ 'execute' => "CREATE DATABASE IF NOT EXISTS $dbname" ] );
 	}
 
+	/**
+	 * Test if the database connection is working.
+	 */
+	public function test_connection() {
+		$sql_result = self::run_sql(
+			'mysql --no-defaults',
+			[
+				'execute'       => 'SELECT 1',
+				'send_to_shell' => false,
+			]
+		);
+		if ( ! empty( $sql_result['stderr'] ) ) {
+			# WP_CLI output functions are suppressed in behat context.
+			echo 'There was an error connecting to the database:' . \PHP_EOL;
+			echo '    ' . trim( $sql_result['stderr'] ) . \PHP_EOL;
+			echo 'run `composer prepare-tests` to connect to the database.' . \PHP_EOL;
+			die( 1 );
```

Comment:
👍 

Maybe we could also suggest using SQLite as an alternative ? Often easiest for new contributors.